internal/ratelimit/ratelimit_test.go

```diff
@@ -0,0 +1,664 @@
+	defaults := DefaultManagerOptions()
+
+	assert.NotNil(t, defaults.Logger)
+	assert.NotNil(t, defaults.Threshold)
+	assert.Equal(t, 0.02, *defaults.Threshold)
+	assert.NotNil(t, defaults.SemaphoreWeight)
+	assert.Equal(t, int64(50), *defaults.SemaphoreWeight)
+	assert.NotNil(t, defaults.MinRequestInterval)
+	assert.Equal(t, 50*time.Millisecond, *defaults.MinRequestInterval)
+	assert.NotNil(t, defaults.CleanupInterval)
+	assert.Equal(t, 30*time.Second, *defaults.CleanupInterval)
+}
+
+func TestNoDoubleSleeping(t *testing.T) {
+	// Create manager with longer minimum interval for testing
+	minInterval := 200 * time.Millisecond
+	manager := NewManager(&ManagerOptions{
+		MinRequestInterval: &minInterval,
+		Threshold:          ptrTo(0.1), // Low threshold to trigger throttling
+	})
+
+	// Set up rate limit info that will trigger throttling
+	manager.mu.Lock()
+	manager.rateLimitInfo = &RateLimitInfo{
+		Limit:     100,
+		Remaining: 5, // Low remaining to trigger throttling
+		Reset:     1, // Short reset time
+	}
+	// Set lastRequestTime to trigger minimum interval delay
+	manager.lastRequestTime = time.Now().Add(-100 * time.Millisecond) // 100ms ago
+	manager.mu.Unlock()
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write([]byte(`{"ok": true}`))
+	}))
+	defer server.Close()
+
+	client := resty.New().SetBaseURL(server.URL)
+	client.OnBeforeRequest(manager.RequestMiddleware)
+	client.OnAfterResponse(manager.ResponseMiddleware)
+
+	start := time.Now()
+	_, err := client.R().Get("/test")
+	elapsed := time.Since(start)
+
+	require.NoError(t, err)
+
+	// Should have slept for throttling delay (which is longer than min interval)
+	// but NOT for both delays added together
+	// Throttling delay calculation:
+	// - resetBuffer = 1 * 0.8 = 0.8s
+	// - baseDelay = 0.8 / 5 = 0.16s
+	// - with 1 active request scaling: 0.16 * 1.2 = 0.192s
+	// - with jitter: 0.192 * (1.1 to 1.2) = 0.211 to 0.230s
+	// Min interval delay would be 100ms (200ms - 100ms already passed)
+	// If we were double sleeping, it would be ~330ms+ (230ms + 100ms)
+	// With the fix, it should be just the throttling delay (~210-230ms)
+
+	assert.Greater(t, elapsed, 200*time.Millisecond, "Should have throttled")
+	assert.Less(t, elapsed, 280*time.Millisecond, "Should not have double slept")
+
+	t.Logf("Request took %v (should be ~210-230ms for throttling only, not ~330ms+ for double sleep)", elapsed)
+}
+
+func TestSemaphoreTimeoutNoReleasePanic(t *testing.T) {
+	// Create manager with debug logging to capture log messages
+	var buf bytes.Buffer
+	logger := slog.New(slog.NewTextHandler(&buf, &slog.HandlerOptions{
+		Level: slog.LevelDebug,
+	}))
+
+	// Create manager with very small semaphore weight to force immediate "timeouts"
+	semaphoreWeight := int64(1)
+	manager := NewManager(&ManagerOptions{
+		SemaphoreWeight: &semaphoreWeight,
+		Logger:          logger,
+	})
```

Comment:
I think you can do stuff like this.
```suggestion
	manager := NewManager(&ManagerOptions{
		SemaphoreWeight: toPtr(1),
		Logger:          logger,
	})
```